R/html.R

```diff
@@ -440,6 +444,40 @@ split_chapters = function(output, build = build_chapter, number_sections, split_
   nms[j]
 }
 
+build_404_page <- function(html_head, html_toc, html_foot, build, ...) {
+  path_404 = "404.html"
+  # if a 404 page already exist, we do nothing specific and assume
+  # user has already a workflow in place
+  if (file.exists(path_404)) return()
```

Comment:
If the 404 page exists, we should also move it via `move_to_output_dir()` above, right? (If you `return()`, it won't be moved)

Reply:
If there is already a 404.html, then it should not be moved I think but it could be copied to output_dir. 
If a 404.html exists, then it should be consider a source file and not moved to output dir (which could be cleaned after that with `clean_book()`, or could be gitignore). Moving would delete right ? 

However, it could be copied. We assumed that if someone has already a 404.html in a bookdown project then there is already a custom way in the project to make this move to the output dir. But it would indeed be more interesting to have this internalize in bookdown. Do you think it is safe to do this ? 

Currently we found one example where a 404 page exists and we did not want to override any mechanism they could have from moving the file. It is for https://github.com/jtr13/EDAV/ but from their source and from https://edav.info/publish.html#make-a-custom-404-page, it is highly possible that the 404 page is copied to output dir just because they linked it in their doc and it is seen as a resource. 

So we could definitely offer to automatically copy to output_dir a 404.html existing in the source folder. 

Does it make sense ?

Reply:
Okay, let's just leave it untouched and assume that if 404.html exists, authors know how to copy it.